src/pages/word-duel.jsx

```diff
@@ -26,70 +19,63 @@ export default function WordDuel() {
 
     function checkGuess(guess) {
 
-        guess = guess.split('');
-
-        if (guess.length !== word.length) {
-            setMessage("Guess wrong length!");
+        if (util.isMaxGuesses(guesses)) {
+            setMessage("Maximum number of guesses already reached.");
+            setGameOver(true);
             return false;
         }
 
-        if (guesses.length >= MAX_GUESSES) {
-            setMessage("Guess limit reached.");
-            setGameOver(true);
+        if (!util.isWordLength(word, guess)) {
```

Comment:
We only call this once, probably could remove the extra function call for it and put the boolean inline here